src/App.tsx

```diff
@@ -1,24 +1,215 @@
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+import {
+  createTodo, deleteTodo, getTodos, updateTodo,
+} from './api/todos';
+import { TodoStatus } from './types/TodoStatus';
+import { ErrorType } from './types/ErrorType';
 
-const USER_ID = 0;
+export const USER_ID = 9929;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<TodoStatus>(TodoStatus.All);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [errorType, setErrorType] = useState<ErrorType>(ErrorType.NoError);
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadedTodoId, setLoadedTodoId] = useState<number[]>([]);
+  const isError = errorType !== ErrorType.NoError;
+
+  const getTodoList = async () => {
+    try {
+      const newTodoList = await getTodos(USER_ID);
+
+      setTodos(newTodoList);
+    } catch {
+      setErrorType(ErrorType.AddError);
+    }
+  };
+
+  useEffect(() => {
+    if (isError) {
+      setTimeout(() => {
+        setErrorType(ErrorType.NoError);
+      }, 3000);
+    }
+  }, [isError, errorType]);
+
+  useEffect(() => {
+    getTodoList();
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    switch (status) {
+      case TodoStatus.Active:
+        return todos.filter(todo => !todo.completed);
+
+      case TodoStatus.Completed:
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  }, [todos, status]);
+
+  const addTodo = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!newTodoTitle.trim()) {
+      setErrorType(ErrorType.EmptyInput);
+
+      return;
+    }
+
+    const newTempTodo = {
+      id: 0,
+      title: newTodoTitle,
+      completed: false,
+    };
+
+    setTempTodo({ ...newTempTodo, userId: USER_ID });
+    setIsLoading(true);
+    setNewTodoTitle('');
+
+    try {
+      const newTodo = await createTodo(USER_ID, newTodoTitle);
+
+      setTodos((prevTodos) => [...prevTodos, newTodo]);
+    } catch {
+      setErrorType(ErrorType.AddError);
+    } finally {
+      setIsLoading(false);
+      setTempTodo(null);
+    }
+  };
+
+  const removeTodo = async (todoId: number) => {
+    try {
+      setLoadedTodoId(state => [...state, todoId]);
+      await deleteTodo(todoId);
+      setTodos((prevTodos) => prevTodos.filter(todo => todo.id !== todoId));
+    } catch {
+      setErrorType(ErrorType.DeleteError);
+    } finally {
+      setLoadedTodoId([]);
+    }
+  };
+
+  const clearCompletedTodos = () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    setLoadedTodoId(completedTodos.map(todo => todo.id));
+
+    completedTodos.forEach(todo => {
```

Comment:
Instead of using forEach with async operations, consider using Promise.all with map to handle multiple async operations efficiently.

Reply:
agree fix it. 
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/97547824/a6b6e399-5ec7-460b-9da0-17c85d9d7980)